packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/documentservice/DictionariesResource.java

```diff
@@ -22,7 +27,31 @@ public class DictionariesResource {
 	@Produces(MediaType.APPLICATION_JSON)
 	@Tag(name = "Get available dictionaries")
 	@Path("/dictionaries")
-	public XDDResponse<XDDDictionariesResponseOK> getAvailableDictionaries(@QueryParam("all") @DefaultValue("") String all) {
-		return proxy.getAvailableDictionaries(all);
+	@APIResponses({
+		@APIResponse(responseCode = "404", description = "An error getting available dictionaries"),
+		@APIResponse(responseCode = "204", description = "Request received successfully, but there are no dictionaries")})
+	public Response getAvailableDictionaries(@QueryParam("all") @DefaultValue("") String all) {
+		try {
+			XDDResponse<XDDDictionariesResponseOK> response = proxy.getAvailableDictionaries(all);
+
+			if (response.getErrorMessage() != null) {
+				Response.status(Response.Status.INTERNAL_SERVER_ERROR)
```

Comment:
I'd prefer [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) than 500, in that case.
What do you think?

Reply:
From our discussion, we're opting to settle on 500 for now